### PR TITLE
Upgrade to 0.12 and add CI

### PR DESCRIPTION
**Pull request by ludoo**
Tests have not been ported yet, so this is still WIP.

### CODE REVIEW

**Comment by ludoo** [comment]
This is now ready, tests have been added for public and private zones.

File: README.md [hunk 1]
```diff
@@ -7,23 +7,26 @@ The resources/services/activations/deletions that this module will create/trigge
...
-  name       = "Foo zone."
-  dns_name   = "foo.local."
-  private_visibility_config = [{
-    networks = [{
-      network_url = "my-vpc"
-    }]
-  }]
+module "dns-private-zone" {
+  source     = "../.."
+  project_id = var.project_id
```
Comment (morgante) [inline README.md]: I recommend *not* using variables in the README example, as they make the module intent less clear. Having hard-coded values is a better example of what each input actually expects.

File: README.md [hunk 2]
```diff
@@ -7,23 +7,26 @@ The resources/services/activations/deletions that this module will create/trigge
...
-  zone_type  = "private"
-  name       = "Foo zone."
-  dns_name   = "foo.local."
-  private_visibility_config = [{
-    networks = [{
-      network_url = "my-vpc"
-    }]
-  }]
+module "dns-private-zone" {
+  source     = "../.."
```
Comment (morgante) [inline README.md]: This should be sourced from the registry + versioned.

Comment (ludoo) [reply README.md]: ahhh thanks, copy/pasted from the example and missed that, will fix asap and then I guess merge and release?

**Pull request merged**